### PR TITLE
XML Config for settings configuration

### PR DESCRIPTION
**Pull request by nmnambiar**
adds xml config
(synced with all my prev PRs)
multiple xml packages were possible (DOM/SAX/StAX, pullAPIs/pushAPIs etc). 
I went with stax because it's said to be memory efficient.

Relevant links
  [xml-tutorial][xml-tuto]
  [nested-xml][xml-nest]

Sample xml file (resources/infrastructure_logger.xml)
```xml
<?xml version="1.0" encoding="UTF-8"?>
<logging>
    <loggerOptions>
        <loggerOption LoggerName="FileLogger"   >false</loggerOption>
        <loggerOption LoggerName="ConsoleLogger">false</loggerOption>
    </loggerOptions>
    <logLevels>
        <logLevel level="ERROR"  >false</logLevel>
        <logLevel level="WARNING">false</logLevel>
        <logLevel level="SUCCESS">false</logLevel>
        <logLevel level="INFO"   >false</logLevel>
    </logLevels>
</logging>
```

TODO: enable/disable logLevels in both fileLogger and consoleLogger.

[xml-tuto]:<https://www.baeldung.com/java-stax> "Parsing an XML File Using StAX"
[xml-nest]:<https://stackoverflow.com/questions/10388047/how-can-i-parse-nested-elements-in-sax-parser-in-java> "Remember parent tag and depth in nested XMLs"

### CODE REVIEW

**Comment by nmnambiar** [comment]
IMPORTANT things to clarify with Architect:
A default XML config file is to be added imo. I propose adding it to a new directory, placed parallel to src and specs, as shown below. This can be accessed from the runtime itself
```
  KoraKaagaz
    |- src
    |     |- infrastructure
    |- specs
    |     |- infrastructure
    |- resources
          |- infrastructure_logger.xml
```

Some relevant links to read up:
  [Read a file from resources folder][file-within-resources]
  [loading such XML files][file-within-src] and [ensuring types][fis-to-is]

[file-within-resources]:<https://mkyong.com/java/java-read-a-file-from-resources-folder/>
[file-within-src]:<https://stackoverflow.com/questions/16141360/read-xml-file-from-within-the-java-project> "xml files within java project"
[fis-to-is]:<https://stackoverflow.com/questions/2353057/how-to-convert-inputstream-to-fileinputstream> "inputStream to fileInputStream"

**Comment by nmnambiar** [comment]
Seeking clarification: how test harness will be running tests

Plan was to have a testMode (enabled by the logger xml)
However, I'm not sure where unit tests will be written
 recent java repos which I've [looked][xml-tuto] [at][tachi-old] seem to follow a structure based on JUnit tests (I think) which would be like this:
 ```
   KoraKaagaz
     ...
     |- src
     |    |- main (the actual stuff)
     |    |    |- java/com/korakaagaz/
     |    |    |- resources
     |    |- test (test cases)
     |    |    |- java/com/korakaagaz/
     |    |    |- resources
     ...  
```

If you havent realised why I'm asking this: the resources folder will be containing the necessary config xml files. This completely changes the location and needs change in what is hardcoded.
Let me know what the team leads, architect has decided.

[xml-tuto]:<https://github.com/eugenp/tutorials/tree/master/xml/src> "XML tutorial on StAX"
[tachi-old]:<https://github.com/inorichi/tachiyomi/tree/dev/app> "tachiyomi (prev in java, now shifted to kotlin)"

**Comment by Rahul-R5** [comment]
> Seeking clarification: how test harness will be running tests
> 
> Plan was to have a testMode (enabled by the logger xml)
> However, I'm not sure where unit tests will be written

> Let me know what the team leads, architect has decided.

Currently the unit tests are decided to be in src/modulename/tests

File: resources/infrastructure_logger.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        <loggerOption LoggerName="FileLogger"   >false</loggerOption>
+        <loggerOption LoggerName="ConsoleLogger">false</loggerOption>
+    </loggerOptions>
+    <logLevels>
+        <logLevel level="ERROR"  >false</logLevel>
+        <logLevel level="WARNING">false</logLevel>
+        <logLevel level="SUCCESS">false</logLevel>
+        <logLevel level="INFO"   >false</logLevel>
+    </logLevels>
+</logging>
```
Comment (Rahul-R5) [inline resources/infrastructure_logger.xml:13]: Add the test mode option and a path to the other config file

File: .classpath [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER">
```
Comment (Rahul-R5) [inline .classpath:5]: Did you ask architect before modifying this file?

**Review by Rahul-R5 — CHANGES_REQUESTED** [review]

File: .classpath (hunk 2 above)
Comment (nmnambiar) [reply .classpath:5]: I've put all comments regarding this at top

**Comment by nmnambiar** [comment]
Closes convict-git#115

File: resources/infrastructure_logger.xml (hunk 1 above)
Comment (nmnambiar) [reply resources/infrastructure_logger.xml:13]: should I make a separate PR for that?

Comment (Rahul-R5) [reply resources/infrastructure_logger.xml:13]: No, edit here itself

File: .classpath (hunk 2 above)
Comment (Rahul-R5) [reply .classpath:5]: Permission received

**Pull request merged**